### PR TITLE
feat(search ui): Adding support for pluggable filter rendering

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

In some cases, using the default way of rendering search filters that is used everywhere : embedded search, search v1, and search v2, does not result in a pleasing or intuitive user experience. Thus, in this PR we introduce a way to customize the rendering of search filters across the app by adding a new concept of "Filter Renderer", which is a plugin allowing us to take over the rendering process for a particular Filter field and do it in a completely custom way across each search surface area. In the future, I expect that we'll migrate much of the hard-codeyness of existing filters into this more general, pluggable rendering system. 

Here I've included the basic framework, plugging into each search surface including

1. An interface for defining a custom Filter Renderer associated with a given field returned from search API
2. A registry for registering custom Filter Renderers, which will be invoked at rendering time if one is found for the given search API field.
3. Some basic unit tests

## Status

Ready for review!

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by joshuaeilers — APPROVED** [review]
Looks great. Only thing that came to mind is how this works for advanced filters. In that case you'd probably only want to allow a value of `true` or just add it without a modal or something like that. Curious, does it work as-is?

File: datahub-web-react/src/app/search/filters/MoreFilterOption.tsx [hunk 1]
```diff
@@ -5,26 +5,10 @@ import React from 'react';
...
-    cursor: pointer;
-
-    &:hover {
-        background-color: ${ANTD_GRAY[3]};
-    }
-
-    ${(props) => props.isActive && `color: ${props.theme.styles['primary-color']};`}
-    ${(props) => props.isOpen && `background-color: ${ANTD_GRAY[3]};`}
-`;
+import { MoreFilterOptionLabel } from './styledComponents';
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/search/filters/MoreFilterOption.tsx:11]: Thank you!!

**Comment by jjoyce0510** [comment]
Advanced filters will be provided in a followup! Thanks for the great review

**Comment by jjoyce0510** [comment]
Running locally success. Blocking important Acryl features.

**Pull request merged**